group-generators/helpers/data-providers/tally/provider.ts

```diff
@@ -0,0 +1,269 @@
+import { gql } from "graphql-request";
+
+import {
+  ITallyProvider,
+  inputGetGovernanceProposers,
+  inputGetProposalVoters,
+  queryData,
+  inputQueryToName,
+} from "./types";
+import { GraphQLProvider } from "@group-generators/helpers/data-providers/graphql";
+import { FetchedData } from "topics/group";
+
+export default class TallyProvider extends GraphQLProvider implements ITallyProvider {
+  url: string;
+  headers: {
+    "Api-Key": string;
+  };
+  public constructor() {
+    super({
+      url: "https://api.tally.xyz/query",
+      headers: {
+        "Api-Key": process.env.TALLY_API_KEY as string,
+        accept: "application/json",
+      },
+    });
+  }
+
+  private async _queryNameToGovernorAddress(): Promise<queryData> {
+    return this.query(
+      gql`
+        query Governors {
+          governors(chainIds: "eip155:1") {
+            id
+            type
+            name
+            proposals {
+              id
+              title
+              description
+              proposer {
+                address
+              }
+              votes {
+                id
+                voter {
+                  id
+                  address
+                }
+              }
+            }
+          }
+        }
+      `,
+      {}
+    );
+  }
+
+  private async _queryNameToGovernorAddressForDelegates(
+    ids: string,
+    offset: number,
+    limit: number
+  ): Promise<queryData> {
+    return this.query(
+      gql`
+        query Governors($ids: [AccountID!], $limit: Int!, $offset: Int!) {
+          governors(chainIds: "eip155:1", ids: $ids) {
+            id
+            name
+            delegates(pagination: { limit: $limit, offset: $offset }) {
+              account {
+                address
+              }
+              stats {
+                voteCount
+              }
+            }
+          }
+        }
+      `,
+      { ids, limit, offset }
+    );
+  }
+
+  private async _queryNameToGovernorId(): Promise<inputQueryToName> {
+    return this.query(
+      gql`
+        query Governors {
+          governors(chainIds: "eip155:1") {
+            id
+            name
+          }
+        }
+      `,
+      {}
+    );
+  }
+
+  public async getProposalVoters({
+    name,
+    proposalId,
+  }: inputGetProposalVoters): Promise<FetchedData> {
+    const data = await this._queryNameToGovernorAddress();
+
+    const fetchedData: { [address: string]: number } = {};
+
+    if (data && Array.isArray(data.governors)) {
+      const matchingGovernors = data.governors.filter(
+        (governor: { name: string }) => governor.name === name
+      );
+
+      for (const governor of matchingGovernors) {
+        for (const proposals of governor.proposals) {
+          const propId = proposals.id;
+
+          if (propId == proposalId) {
```

Comment:
Can we keep the strict equality operator `===` instead of the equality operator `==` please?
It forces type checking and it is used across the codebase 

I let you update in the other parts where the equality operator is used

Reply:
changed the same